### PR TITLE
Refactor folder structure and change file names

### PR DESCRIPTION
**Pull request by Dennis-Bitoiu**
- Add new folder structure
- Rename files
- Rename Types/Classes
- Add comments

### CODE REVIEW

File: src/app/core/transaction-mapper/impl/revolut-transaction-mapper.ts
Comment (eduard-cc) [inline src/app/core/transaction-mapper/impl/revolut-transaction-mapper.ts:19]: I agree,

but InputTransactionCreate type is used here to be able to call class-validator's validate(), since InputTransactionCreate has all the validation decorators

can maybe move the class-validator decorators to the Transaction ObjectType or Entity?

File: src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts
Comment (eduard-cc) [inline src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts:23]: Idk what max value to use, but it should be a high value anyway and be consistent everywhere

maybe Number.MAX_SAFE_INTEGER

also check out https://v2.dinerojs.com/docs (or another equivalent) for expressing monetary values, will probably be useful later

File: .vscode/settings.json
Comment (eduard-cc) [inline .vscode/settings.json:1]: is this file necessary to be commited?

File: src/app/entities/user/user.entity.ts
Comment (eduard-cc) [inline src/app/entities/user/user.entity.ts:4]: I thought class validator should be used for incoming data like InputType or ArgsType, not entities

since name, email and password are already validated at this point, and createdDate, isActive and id are added by the DB

**Review by eduard-cc — APPROVED** [review]

**Comment by eduard-cc** [comment]
Approved, TODO after merging:

- add @ alias for absolute paths
- rewrite / fix tests

File: src/app/core/transaction-mapper/impl/revolut-transaction-mapper.ts
Comment (Dennis-Bitoiu) [reply src/app/core/transaction-mapper/impl/revolut-transaction-mapper.ts:19]: I will move the class validator decorators to the entity.

File: src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts
Comment (Dennis-Bitoiu) [reply src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts:23]: But is the `@Max` decorator even needed? 
I mean I don't think a transaction of 999999999999$ will ever take place

File: .vscode/settings.json
Comment (Dennis-Bitoiu) [reply .vscode/settings.json:1]: No, I thought it was in the `.gitignore`.

File: src/app/api/graphql/modules/users/graphql/user.queries.resolver.e2e.spec.ts
Comment (eduard-cc) [inline src/app/api/graphql/modules/users/graphql/user.queries.resolver.e2e.spec.ts:1]: why is this file named e2e? i believe these are unit tests not end to end tests

File: src/app/entities/user/user.entity.ts
Comment (Dennis-Bitoiu) [reply src/app/entities/user/user.entity.ts:4]: I think we should add the validation on the Database layer rather than the GraphQL one

File: .vscode/settings.json
Comment (Dennis-Bitoiu) [reply .vscode/settings.json:1]: But it would be nice if we all would have the same settings

File: src/app/core/auth/auth.service.spec.ts
Comment (eduard-cc) [inline src/app/core/auth/auth.service.spec.ts:65]: I would avoid using skip to ignore tests, let them fail, pushing to develop doesn't currently require CI to pass

File: src/app/api/graphql/modules/users/graphql/user.queries.resolver.e2e.spec.ts
Comment (Dennis-Bitoiu) [reply src/app/api/graphql/modules/users/graphql/user.queries.resolver.e2e.spec.ts:1]: Probably I renamed it by mistake

File: src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts
Comment (eduard-cc) [reply src/app/api/graphql/modules/transactions/graphql/transaction-create.input.ts:23]: would probably be an edge case, if parsing a file goes wrong

also, if users will be allowed to manually add transactions in the future

File: src/app/core/auth/auth.service.spec.ts
Comment (Dennis-Bitoiu) [reply src/app/core/auth/auth.service.spec.ts:65]: I'd rather use skip. That way the pipeline still succeeds and  and these `skips` are easy to find and fix once we add the required things.

File: package.json
Comment (eduard-cc) [inline package.json:13]: what is this for? seems to use rimraf dependency, if so npm install it

**Pull request merged**
